src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+data ExistsRE (f :: # ! -> *)
```

Comment:
There are two ways, depending on your perspective. The first perspective is that every empty type literally *is* False, so if you can construct an inhabitant of it, you have a proof of falsehood.

Secondly, sum types are *always* meant to support pattern matching, even the special case of the nullary sum. So, even if your language doesn't have syntax for pattern matching on a nullary sum, the below is *always* a valid thing to do, assuming that you know that `Empty` is a sum type:

```
absurd :: forall a. Empty -> a
absurd x = unsafeCoerce "the pattern `x` is impossible!"
```

Now, on the contrary in (a hypothetical version of) ML (where you can declare a nullary sum) the following two signatures are quite different: 

```
signature SIG1 = 
sig
  type welp
end

sig SIG2 =
sig
  datatype welp = .
end
```

In `SIG1`, `welp` is a type about which you know nothing: you can neither construct it nor destruct it. On the other hand, in `SIG2`, `welp` is the empty type and is isomorphic to `false`. So, you know that it can have no values ever.